### PR TITLE
build: add backport script

### PR DESCRIPTION
**Pull request by Harjot1Singh**
## Summary

Adds a backporting script to 4.x releases from our next branch, and CI integration for the process.

See `backport.js` for a description of what transformations are applied.

Note - this is currently being tested on my fork.

### CODE REVIEW

File: lib/backport.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+const JSON_PATH = resolve( __dirname, '../data' )
+
+const GURMUKHI_SOURCES = {
+  'Ardaas': 'SGPC',
+  'Ganj Nama Bhai Nand Lal Ji': 'Dr. Ganda Singh',
+  'Ghazals Bhai Nand Lal Ji': 'Dr. Ganda Singh',
+  'Jot Bigas Bhai Nand Lal Ji': 'Dr. Ganda Singh',
+  'Kabit Savaiye Bhai Gurdas Ji': 'Seva Singh',
+  'Sarabloh Granth': 'Amrit Keertan',
+  'Sri Dasam Granth': 'SGPC',
```
Comment (Sarabveer) [inline lib/backport.js:26]:
```suggestion
  'Sri Dasam Granth': 'SGPC' || 'Budha Dal Mehron',
```

SGPC doesn't span the whole composition, so a fallback source is needed. Not sure if this is the proper way to do it.

**Review by Sarabveer — CHANGES_REQUESTED** [review]

File: lib/backport.js (hunk 1 above)
Comment (Harjot1Singh) [reply lib/backport.js:25]: Very good call out, this would not be the way to do it, but I'll update my PR

Comment (Harjot1Singh) [reply lib/backport.js]: Are there any other cases of this?

Comment (Sarabveer) [reply lib/backport.js]: Let me check

Comment (Sarabveer) [reply lib/backport.js]: LGTM

**Review by Sarabveer — DISMISSED** [review]

**Review by Sarabveer — APPROVED** [review]
LGTM

**Review by bhajneet — APPROVED** [review]

**Pull request merged**